File: Python/6py 1/dig_pow.py
"""
|--------------------------------------------------------------------------
| Playing with digits
|--------------------------------------------------------------------------
|
"""

def dig_pow(n, p):
   res = 0
   for i in range(len(str(n))):
      res += int(str(n)[i]) ** (p + i)
   return res // n if res % n == 0 else -1


def dig_pow(n, p):
   t = sum( int(d) ** (p+i) for i, d in enumerate(str(n)) )
   return t//n if t%n==0 else -1


def dig_pow(n, p):
   k, fail = divmod(sum(int(d)**(p + i) for i, d in enumerate(str(n))), n)
   return -1 if fail else k


print(dig_pow(89, 1), 1)
print(dig_pow(92, 1), -1)
print(dig_pow(46288, 3), 51)
print(dig_pow(41, 5), 25)
print(dig_pow(114, 3), 9)
print(dig_pow(8, 3), 64)

File: PHP/6php 1/digPow.php
<?php
/*
|--------------------------------------------------------------------------
| Playing with digits
|--------------------------------------------------------------------------
|
*/

function digPow($n, $p) {
   $res = 0;
   for ($i = 0; $i < strlen(strval($n)); $i++) {
      $res += intval(strval($n)[$i]) ** ($p + $i);
   }
   return $res % $n === 0 ? intval($res / $n) : -1;
}


function digPow($n, $p) {
   $res = array_reduce(str_split($n), function($acc, $curr) use (&$p) {
      return $acc + $curr ** $p++;
   }, 0) / $n;
   return is_int($res) ? $res : -1;
}


function digPow($n, $p) {
   $sum = 0;
   foreach (str_split($n) as $i => $e) {
      $sum += $e ** ($i + $p);
   }
   return $sum % $n ? -1 : $sum / $n;
}


print_r(digPow(89, 1) . ' ' . 1 . '
');
print_r(digPow(92, 1) . ' ' . -1 . '
');
print_r(digPow(46288, 3) . ' ' . 51 . '
');